.github/workflows/test.yml

```diff
@@ -0,0 +1,55 @@
+name: xpra
+
+on: [push, pull_request]
+
+jobs:
+  unittest:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: "apt-get install"
+      run: |
+        # see docs/Build/Debian.md
+        sudo apt-get update
+        sudo apt-get -y install libx11-dev libxtst-dev libxcomposite-dev libxdamage-dev libxres-dev \
+                libxkbfile-dev python-all-dev pandoc libsystemd-dev \
+                libgtk-3-dev python3-dev python3-cairo-dev python-gi-dev cython3 \
+                xauth x11-xkb-utils \
+                xvfb python3-cairo python3-gi-cairo \
+                python3-opengl python3-lz4 python3-rencode python3-pil \
+                libturbojpeg-dev libwebp-dev python3-pil \
+                libx264-dev libvpx-dev yasm \
+                python3-numpy \
+                libavformat-dev libavcodec-dev libswscale-dev \
+                python3-opengl \
+                python3-rencode python3-lz4 python3-dbus python3-cryptography \
+                python3-netifaces python3-yaml python3-lzo \
+                openssh-client sshpass python3-paramiko \
+                python3-setproctitle python3-xdg python3-pyinotify python3-opencv \
+                libpam-dev quilt xserver-xorg-dev xutils-dev xserver-xorg-video-dummy xvfb keyboard-configuration \
+                python3-kerberos python3-gssapi \
+                gstreamer1.0-pulseaudio gstreamer1.0-alsa \
+                gstreamer1.0-plugins-base gstreamer1.0-plugins-good \
+                gstreamer1.0-plugins-ugly \
+                cups-filters cups-common cups-pdf python3-cups \
+                liblz4-dev \
+                python3-coverage \
+                x11-xserver-utils dbus-x11
+
+        # tests/unittests/run wants "coverage", but debian installs as /usr/bin/python3-coverage
+        sudo ln -s /usr/bin/python3-coverage /usr/local/bin/coverage
+
+    - name: "Build/Install/Test"
+      run: >
+        python3 setup.py unittests
+        --skip unit.x11.x11_server_test
+        --skip unit.client.splash_test
+        --skip unit.client.x11_client_test
+        --skip unit.server.mixins.start_option_test
+        --skip unit.server.mixins.shadow_option_test
+        --skip unit.server.mixins.startdesktop_option_test
+        --skip unit.server.server_auth_test
+        --skip unit.server.server_sockets_test
+        --skip unit.scripts.main_test
+        --skip unit.codecs.pillow_encoder_test
```

Comment:
Do we have a record of why each one of these tests is disabled?
We should work towards running them all, so having this information would help.

Reply:
The pillow test is fixed: 6f28b0bd28f36a5eb171989967e45da57295bbfc

Reply:
`--skip-fail` vs. `--skip-slow` should record why a test is ignored/skipped.